src/vs/workbench/contrib/debug/common/debugSchemas.ts

```diff
@@ -127,6 +127,31 @@ export const breakpointsExtPoint = extensionsRegistry.ExtensionsRegistry.registe
 	}
 });
 
+export interface IDisassemblyContribution {
+	language: string;
+}
+
+// disassembly extension point
+export const disassemblyExtPoint = extensionsRegistry.ExtensionsRegistry.registerExtensionPoint<IDisassemblyContribution[]>({
```

Comment:
@xisui-MSFT what's the purpose of this contribution point?

Reply:
My understanding is that `supportsDisassembleRequest` is set when the debugger supports disassemble request? However, we don't want the "show disassembly"/"go to disassembly" option to exist in files other than those of certain languages.

Ultimately, our goal is to achieve similar functions as in VS. I.e., when right clicking a certain line in source (may not be %rip), go to disassembly corresponds to that line.

Reply:
@xisui-MSFT Thanks, now I understand the problem you are addressing with the contribution point.

I think for now it is OK to introduce an (internal) contribution point.
But since a general solution will involve public extension API, we'll have to consider more scenarios in different debuggers before we can create a "proposed API" (that will have to go through the regular VS Code API process).

Reply:
Alternative idea: let's not introduce a new contribution point but instead use the `languages` contribution point which debug extension already use to say they are interested in some `langauges`. [Mock debug example](https://github.com/microsoft/vscode-mock-debug/blob/main/package.json#L135).

So the logic would be, if there is an active debug session which `supportsDisassembleRequest` then show the action for all editor language types that are specified under the languages contribution of that debug extension.

What do you think?

Reply:
Yes, adding a `disassemblyLanguages` property to the `debuggers` contribution points is an alternative way to address the problem in a **static** way. We'll have to see whether this is enough...

Reply:
Just to clarify: I am arguing that we should use the existing `languages` contribution point that we already have :)

Reply:
Yeah, but only as a temporary measure, not as official API.

We cannot extend the semantics of the `languages` contribution point to cover languages that can be disassembled.
Introducing a similar property `disassemblyLanguages` would keep them separate.

Reply:
Where would be a good point to initialize this ext point? Looks like AdapterManager is usually accessed via IDebugService, but we can't use IDebugService in the precondition of [OpenDisassemblyViewAction](https://github.com/microsoft/vscode/blob/180c83aadda613f17ebba817c06c350f29057946/src/vs/workbench/contrib/debug/browser/debugEditorActions.ts#L137), since its constructor requires an empty parameter list.

Also, how is `CONTEXT_DISASSEMBLE_REQUEST_SUPPORTED` supposed to be used? Should it be set by the debugger adapter or the extension?

Reply:
CONTEXT_DISASSEMBLE_REQUEST_SUPPORTED shuold be set [here](https://github.com/microsoft/vscode/blob/main/src/vs/workbench/contrib/debug/common/debugViewModel.ts#L36)
based on the focused session

For the new ext point as we discussed above this is not needed for step one, simply use the `debuggers.languages` contribution point for now.

Reply:
I'm fine with either way... My point is that looks like we can't use IDebugService in the precondition of [the action](https://github.com/microsoft/vscode/blob/180c83aadda613f17ebba817c06c350f29057946/src/vs/workbench/contrib/debug/browser/debugEditorActions.ts#L137), but seems like IDebugService is the only way to access this ext point.

Reply:
@xisui-MSFT define the precondition in `common/debug.ts` as all other preconditiions are defined. And they you will be able to access it from the action. And the debugViewModel will actually "implement" the precondition based on focused session.

Reply:
I guess you meant to make it a context key just like `CONTEXT_DISASSEMBLE_REQUEST_SUPPORTED`? Although I can change debugAdapterManager to be able to access the ext point, I'm not sure how to get the language id of the current editor in it?

Reply:
Using workbench contribution as we discussed offline.